middleware/dnstap/msg/msg.go

```diff
@@ -87,3 +95,29 @@ func (d *Data) ToClientQuery() *tap.Message {
 		QueryPort:      &d.Port,
 	}
 }
+
+// Transform the data into a forwader or resolver query message.
```

Comment:
Does this need to be exported? Also use `go vet` and `golint` to find lint violation about functions docs.